tendrl/gluster_integration/objects/volume/atoms/create/__init__.py

```diff
@@ -41,6 +41,43 @@ def run(self):
                 "force": self.parameters.get('Volume.force')
             })
 
+        # mark the bricks locked by the task_id so that they cannot be
+        # picked by other create volume tasks
+        for sub_vol in self.parameters.get('Volume.bricks'):
+            for brick in sub_vol:
+                ip = brick.keys()[0]
+                brick_path = brick.values()[0]
+                try:
+                    # Find node id using ip
+                    node_id = NS._int.client.read("indexes/ip/%s" % ip).value
+                except etcd.EtcdKeyNotFound:
+                    # Find node id using hostname
+                    nodes = NS._int.client.read("nodes/")
+                    for node in nodes.leaves:
+                        hostname = NS._int.client.read(
+                            "%s/NodeContext/fqdn" % node.key).value
+                        if hostname == ip:
+                            node_id = node.key.split("/")[-1]
+                key = "nodes/%s/NodeContext/fqdn" % node_id
+                host = NS._int.client.read(key).value
```

Comment:
@GowthamShanmugam so the volume creation request can have brick details in below two forms

```
"Volume.bricks": [
            [
                {"12.34.45.56":"/mnt/sample5/sampleBrick1"},
                {"12.34.45.56":"/mnt/sample4/sampleBrick1"}
            ]
],
```

or 

```
"Volume.bricks": [
            [
                {"fqdn1":"/mnt/sample5/sampleBrick1"},
                {"fqdn2":"/mnt/sample4/sampleBrick1"}
            ]
],
```

Suppose scenario-1 is followed and IP is provided. In that case we look at `/indexes/ip/{ip}` and get the node_id from there.

If scenario-2 was followed the key `/indexes/ip/{fqdn1}` will not be found and key not found exception occurs. In that case I loop through the list of nodes and wherever the node fqdn matches, that node's node_id I record to use.

I dont any issue in this logic.